### PR TITLE
Bug fix - Per day costs

### PR DESCRIPTION
**Pull request by SEAJang**
# Pull Request Key Information

## Reason for change

Per day costs were being added daily, even when no deployments were happening. 

## What was changed

Fixed such that deployments impact the per day costs. 

## Intended Purpose

Fix bug related to per day costs not working as intended.

## Level of version change required

Patch

## Testing Completed

Added new unit tests
All existing unit tests pass.
[results.txt](https://github.com/user-attachments/files/16075202/results.txt)

Manually ran simple_test_case1, with a method that had a per_day cost set, and one without to ensure that both are working as intended. 
image

### CODE REVIEW

File: LDAR_Sim/src/programs/method.py [hunk 1]
```diff
@@ -219,16 +219,10 @@ def deploy_crews(self, workplan: Workplan, weather, daylight) -> CrewDeploymentS
         for crew in self._crew_reports:
             # TODO : if method is daylight sensitive, check for max daylight
             crew.day_time_remaining = day_time_remaining
+            crew.deployed = False
             priority_queue.put((-crew.day_time_remaining, crew.crew_id, crew))
 
-        # If the cost type for the method is per day, calculate the deployment cost for day
-        # based off the number of crews being deployed
-        if self.cost_type == self.PER_DAY_COST:
-            if self._deployment_type == pdc.Deployment_Types.STATIONARY:
```
Comment (ThomasGalesloot) [inline LDAR_Sim/src/programs/method.py:227]: Have we tested this with stationary? The case for stationary still holds true, but if new functionality work too I see no issue

File: LDAR_Sim/src/programs/method.py [hunk 2]
```diff
@@ -219,16 +219,10 @@ def deploy_crews(self, workplan: Workplan, weather, daylight) -> CrewDeploymentS
...
-                deploy_stats.deployment_cost = self.cost * len(workplan.site_survey_planners)
-            else:
-                deploy_stats.deployment_cost = self.cost * len(self._crew_reports)
-
+        workplans_to_add = []
```
Comment (ThomasGalesloot) [inline LDAR_Sim/src/programs/method.py:225]: Nit: This needs a type annotation

Comment (ThomasGalesloot) [reply LDAR_Sim/src/programs/method.py:225]: Style and clarity: Also I think renaming to something like unfinished surveys might be more clear

File: LDAR_Sim/src/programs/component_level_method.py [hunk 3]
```diff
@@ -204,4 +202,10 @@ def deploy_crews(self, workplan: Workplan, weather, daylight) -> CrewDeploymentS
             workplan.add_survey_report(survey_report, survey_plan)
```
Comment (ThomasGalesloot) [inline LDAR_Sim/src/programs/component_level_method.py:202]: Problem: Should add the survey reports to the workplan after the loop here like we are doing in method.py - Perhaps a copy of the unit tests for method.py needs to be transferred and exist for component_level_method as well in order to catch things like this

File: LDAR_Sim/testing/unit_testing/test_programs/test_method/method_testing_fixtures.py [hunk 4]
```diff
@@ -641,3 +664,299 @@ def create_random_state():
...
+            6,
+            date(2023, 1, 1),
+            date(2025, 12, 31),
+            [2023, 2024, 2025],
+            [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12],
+        )
+        for site in mock_sites
+    ]
+    workplan = Workplan(survey_plans, date=date(2023, 1, 1))
+    expected_deployment_stats = CrewDeploymentStats(5.0, 1, 2, 120)
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_programs/test_method/method_testing_fixtures.py:851]: Reading the test data I think this should be 10.0, 5, 10, 600 should it not? As you have 5 sites?

Comment (ThomasGalesloot) [reply LDAR_Sim/testing/unit_testing/test_programs/test_method/method_testing_fixtures.py:851]: With each site being 2 hours to survey, 1 crew shouldn't be able to survey all 5 sites in 8 hours

File: LDAR_Sim/testing/unit_testing/test_programs/test_method/test_deploy_crew_costs.py [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+from src.programs.method import Method
+from testing.unit_testing.test_programs.test_method.method_testing_fixtures import (  # noqa
+    deploy_crews_testing3_fix,
+    deploy_crews_testing4_fix,
+    deploy_crews_testing5_fix,
+)
+from src.scheduling.workplan import Workplan
+
+
+def test_000_simple_single_crew_daily_cost(deploy_crews_testing3):
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_programs/test_method/test_deploy_crew_costs.py:31]: Under the test definitions can you briefly describe key inputs, expected outputs and the point of the test for each test, given that the test data is obfuscated by fixtures.

**Review by ThomasGalesloot — CHANGES_REQUESTED** [review]
Great work - Please address the few comments/concerns

File: LDAR_Sim/src/programs/method.py (hunk 1 above)
Comment (SEAJang) [reply LDAR_Sim/src/programs/method.py:227]: Updated for stationary, as stationary by default only has one crew.

**Comment by SEAJang** [comment]
Unit test results (all pass)
[results.txt](https://github.com/user-attachments/files/16088253/results.txt)
Manually tested stationary deployment costs, and mobile per day deployment costs, and ensured that the per site costs were not affected.

File: LDAR_Sim/testing/unit_testing/test_programs/test_method/test_deploy_crew_costs.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+    deploy_stats: CrewDeploymentStats = method.deploy_crews(workplan, weather, daylight)
+
+    assert deploy_stats == expected_deployment_stats
+
+
+def test_000_multi_site_single_crew_daily_cost(deploy_crews_testing4):
+    """
+    This tests for the case where there is a single crew deployed to multiple sites.
+    4 mocked sites
+    1 crews
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply LDAR_Sim/testing/unit_testing/test_programs/test_method/test_deploy_crew_costs.py:51]: the _estimate_method_crews_required returns 1.

**Comment by SEAJang** [comment]
[results.txt](https://github.com/user-attachments/files/16090036/results.txt)

Manually tested again. 
image
image

**Review by ThomasGalesloot — APPROVED** [review]
Great work - Since we have 2 places in the code with this problem can you transfer the new test for method to also exist for component level method before merging

**Comment by SEAJang** [comment]
All unit tests pass
[results.txt](https://github.com/user-attachments/files/16112694/results.txt)

Manually tested again
image

File: LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py [hunk 7]
```diff
@@ -0,0 +1,513 @@
...
+
+
+def mock_false_survey_site(self, crew, survey_report, site_to_survey, weather, curr_date):
+    return SiteSurveyReport(1), 0, False, 0
+
+
+def mock_survey_site(self, crew, survey_report, site_to_survey, weather, curr_date):
+    crew.day_time_remaining -= 120
+    return (
+        SiteSurveyReport(
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py:91]: Can you construct objects with keywords for future proofing

File: LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py [hunk 8]
```diff
@@ -0,0 +1,513 @@
...
+        ),
+        1,
+        True,
+        1,
+    )
+
+
+def mock_survey_site2(self, crew, survey_report, site_to_survey, weather, curr_date):
+    crew.day_time_remaining -= 120
+    return (
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py:114]: Can you construct objects with keywords for future proofing

File: LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py [hunk 9]
```diff
@@ -0,0 +1,513 @@
...
+            "test_method",
+        ),
+        1,
+        False,
+        1,
+    )
+
+
+def mock_get_detectable_emissions(method_name):
+    return {
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py:137]: Can you construct objects with keywords for future proofing

File: LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py [hunk 10]
```diff
@@ -0,0 +1,513 @@
...
+        },
+    }
+    weather = create_random_state()
+    sites = [site_mock]
+    mocker.patch.object(DaylightCalculatorAve, "__init__", mock_daylight_calculator_ave_init)
+    daylight: DaylightCalculatorAve = DaylightCalculatorAve()
+
+    survey_plans = [
+        ScheduledSurveyPlanner(
+            site,
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py:209]: Can you construct objects with keywords for future proofing

File: LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py [hunk 11]
```diff
@@ -0,0 +1,513 @@
...
+            pdc.Method_Params.PER_DAY: 5,
+        },
+    }
+    weather = create_random_state()
+    mocker.patch.object(DaylightCalculatorAve, "__init__", mock_daylight_calculator_ave_init)
+    daylight: DaylightCalculatorAve = DaylightCalculatorAve()
+
+    survey_plans = [
+        ScheduledSurveyPlanner(
+            site,
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py:296]: Can you construct objects with keywords for future proofing

File: LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py [hunk 12]
```diff
@@ -0,0 +1,513 @@
...
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py:386]: Can you construct objects with keywords for future proofing

File: LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py [hunk 13]
```diff
@@ -0,0 +1,513 @@
...
+            6,
```
Comment (ThomasGalesloot) [inline LDAR_Sim/testing/unit_testing/test_programs/test_component_level_method/component_method_testing_fixtures.py:479]: Can you construct objects with keywords for future proofing

**Review by ThomasGalesloot — APPROVED** [review]
Great work!

**Comment by SEAJang** [comment]
All unit tests pass
[results.txt](https://github.com/user-attachments/files/16121936/results.txt)

Manual testing looked good.

**Pull request merged**